File: c++/menuDisciplinasPagas.cpp

vector<disciplina> disciplinasPagas;

void adicionaDscPaga(string nome, vector<disciplina>& dscPagas) {
    if(temDisciplina(nome)){
        disciplina dscPaga = buscarDisciplina(nome);
        dscPagas.push_back(dscPaga);
    }
}
    

void removeDscPaga(string nome, vector<disciplina>& dscPagas) {
    int i = 0;
    vector<disciplina>::iterator it = dscPagas.begin();
    for(it; it != dscPagas.end(); it++){
        if((*it).nome == nome) {
            dscPagas.erase(disciplinasPagas.begin() + i);
            break;
        }
        i++;
    }
}

void informaDisciplinasPagas() {
    system("clear");
    
    cout << "Menu de disciplinas pagas" << endl << endl;
    string entrada = "";
    string comando = "";

    //Leitura de quais disciplinas o usuário já pagou
    cout << "Quais disciplinas voce ja pagou?       | Digite \"pronto\" quando terminar \n"; 
    cout << ">> ";
    cin >> entrada;
    transform(entrada.begin(), entrada.end(), entrada.begin(), ::toupper);

    while(entrada != "PRONTO"){
        
        if(temDisciplina(entrada)) {
            adicionaDscPaga(entrada, disciplinasPagas);
        } else{
            cout << CRED << "ERRO! " << RESTORE_COLOR  << "Disciplina não encontrada" << endl;
        }
        cout << ">> ";
        cin >> entrada;
        transform(entrada.begin(), entrada.end(), entrada.begin(), ::toupper);  
    }

    //Remove disciplinas que o usuário possa ter informado erroneamente
    while(comando != "n"){
        cout << "Deseja remover alguma disciplina que informou como paga? s/n ou visulaizar disciplinas pagas(v):\n";
        cout << ">> ";
        cin >> comando;
        if(comando == "v"){
            cout << "disciplinas pagas: " << CGREEN << endl;
            for(disciplina d: disciplinasPagas){
                cout << "         "<< d.nome << endl;
            }
            cout << RESTORE_COLOR <<  endl;
        }
        if(comando == "s") {
            cout << "Qual disciplina deseja remover? ";
            cout << ">> ";
            cin >> entrada;
            removeDscPaga(entrada, disciplinasPagas);
        }
    }

    cout << "Você pagou as seguintes disciplinas.\n\n";
    vector<disciplina>::iterator it = disciplinasPagas.begin();
    for(it; it != disciplinasPagas.end(); it++) {
        cout << (*it).toString();
        cout << "\n";
    }
}

File: haskell/MenuDisciplinasPagas.hs
module MenuDisciplinasPagas (menuPagas) where
    import DisciplinasPagas
    import BancoDisciplinas
    import FuncoesAuxiliares

    menuPagas :: IO()
    menuPagas = do
        putStr ("\ESC[2J")
        putStrLn ("Menu de disciplinas pagas\n")
        putStrLn ("Quais disciplinas voce ja pagou?       | Digite \"pronto\" quando terminar")
        guardarPagas

    
    guardarPagas :: IO()
    guardarPagas = do
        disc <- getLine
        if ((upper disc) == "PRONTO") then  removerPagas
            else do
                if(existeDisciplina disciplinas (upper disc)) then do
                    adicionarDisciplina disc
                    guardarPagas
                else do
                    putStrLn ("ERRO! Disciplina não encontrada")
                    guardarPagas
    
    
    removerPagas :: IO()
    removerPagas = do
        pagas <- pegarDiscPagas
        putStrLn ("Deseja remover alguma disciplina que informou como paga? s/n, visualizar disciplinas pagas(v) ou apagar histórico(a):")
        comando <- getLine
        if(comando == "n") then do
            putStrLn "Você pagou as seguintes disciplinas:"
            putStrLn (toStringPagas pagas)
            retornoMenuPrincipal
        else if (comando == "v") then do
            putStrLn ("disciplinas pagas: ")
            putStrLn (toStringPagas pagas)
            removerPagas
        else if (comando == "a") then do
            limpaArquivo pagas
            retornoMenuPrincipal
            
        else do
            putStrLn ("Qual disciplina deseja remover? ")
            disc <- getLine
            removerDisciplina disc
            removerPagas
            
    retornoMenuPrincipal :: IO()
    retornoMenuPrincipal = do
        putStrLn ("\nDeseja retornar ao menu principal? s/n")
        c <- getLine
        if(c == "s") then do 
            putStr ("\ESC[2J")
            return()
            else menuPagas